aws-ec2-transitgatewayroutetable/src/main/java/software/amazon/ec2/transitgatewayroutetable/BaseHandlerStd.java

```diff
@@ -83,32 +83,29 @@ protected ProgressEvent<ResourceModel, CallbackContext> handleError(
             final ResourceModel resourceModel,
             final CallbackContext callbackContext,
             final Logger logger
-           ) {
+    ) {
 
-        BaseHandlerException ex = null;
+        BaseHandlerException ex ;
 
         logger.log(String.format("Handling Error %s for Request [%s] during operation %s", e.toString(), request.toString(), this.getClass().getSimpleName()));
         logger.log(Arrays.toString(e.getStackTrace()));
-        if (UNAUTHORIZED_OPERATION.equals(getErrorCode(e))) {
+        if (UNAUTHORIZED_OPERATION.equals(getErrorCode(e)) || ACCESS_DENIED_ERROR_CODE.equals(getErrorCode(e))) {
             ex = new CfnAccessDeniedException(e);
-        } else if (INVALID_TRANSIT_GATEWAY_STATE.equals(getErrorCode(e))) {
+        } else if (INVALID_TRANSIT_GATEWAY_STATE.equals(getErrorCode(e))
+                || INVALID_REQUEST.equals(getErrorCode(e))  || INVALID_PARAMETER_VALUE.equals(getErrorCode(e))) {
             ex = new CfnInvalidRequestException(e);
-        } else if(INVALID_REQUEST.equals(getErrorCode(e))){
-            ex = new CfnInvalidRequestException(e);
-        }
-        else if (INVALID_PARAMETER_VALUE.equals(getErrorCode(e))) {
-            ex = new CfnInvalidRequestException(e);
-        } else if (INVALID_TRANSIT_GATEWAY_ID.equals(getErrorCode(e))) {
+        }  else if (INVALID_TRANSIT_GATEWAY_ID.equals(getErrorCode(e)) || INVALID_TRANSIT_GATEWAY_ATTACHMENT_ID_NOT_FOUND.equals(getErrorCode(e)) || INVALID_ASSOCIATION_NOT_FOUND.equals(getErrorCode(e)) || INVALID_ID.equals(getErrorCode(e)) || INVALID_ROUTE_TABLE_ID_NOT_FOUND.equals(getErrorCode(e)) || INVALID_ROUTE_NOT_FOUND.equals(getErrorCode(e))) {
             ex = new CfnNotFoundException(e);
-        } else if (INVALID_ID.equals(getErrorCode(e))){
-            ex = new CfnNotFoundException(e);
-        } else if (TRANSIT_GATEWAY_LIMIT_EXCEEDED.equals(getErrorCode(e))) {
+        }  else if (TRANSIT_GATEWAY_LIMIT_EXCEEDED.equals(getErrorCode(e))) {
             ex = new CfnServiceLimitExceededException(e);
         } else if (AUTH_FAILURE.equals(getErrorCode(e))) {
             ex = new CfnInvalidCredentialsException(e);
-        } else if (THROTTLING.equals(getErrorCode(e))){
+        } else if (THROTTLING.equals(getErrorCode(e))) {
             ex = new CfnThrottlingException(e);
-        } else {
+        } else if (INCORRECT_STATE.equals(getErrorCode(e)) || INVALID_ROUTE_TABLE_ID_MALFORMED.equals(getErrorCode(e))) {
```

Comment:
Malformed route table ID exceptions should throw `CfnInvalidRequestException` instead, right? I'm not clear on why this should throw a not stabilized exception.

Reply:
Thank you for the comment i'll fix that !